src/Microsoft.OpenApi/Reader/OpenApiModelFactory.cs

```diff
@@ -320,11 +320,21 @@ private static ReadResult InternalLoad(MemoryStream input, string format, OpenAp
                 }
                 else
                 {
-                    format = Path.GetExtension(url).Split('.').LastOrDefault();
-
                     try
                     {
-                        var fileInput = new FileInfo(url);
+                        string fileName;
+                        if (url.StartsWith("file:", StringComparison.OrdinalIgnoreCase))
+                        {
+                            fileName = new Uri(url).LocalPath;
+                        }
+                        else
+                        {
+                            fileName = url;
+                        }
+
+                        format = null;
```

Comment:
```suggestion
                        format = Path.GetExtension(fileName).Split('.').LastOrDefault();;
```

I'm sorry, I think I go confused by the github diff view last time and missed the fact we already had similar code in place.

Reply:
I think the fact that it was there in the first place is debatable, but it's orthogonal to your change, which is why removing it now does not make a lot of sense. And yet, that still works without it because of the format inspection (reading the stream). That inspection is "more accurate" but has performance impacts. Thanks for reverting the revert.